Combine multiple `has_sync_flag` calls to minimize atomic operations

dipinhora opened the pull request:
We currently make multiple calls to `has_sync_flag` in order to check
if an actor has some flags set or not. Compilers generally do not
optimize `atomic` operations very aggressively and it is very likely
that subsequent calls to `has_sync_flag` all end up executing one
after another instead of being coalesced into a single atomic operation.

This commit combines multiple calls to `has_sync_flag` into a
single call to `has_sync_flag_any` in order to only do a single
atomic read instead of multiple atomic reads. The 
`has_sync_flag_any` function was added for clarity.
The `is_muted` function is removed as part of this for consistency and to
minimize confusion because in some cases a direct call to `has_sync_flag`
had to be made to be able to combine atomic operations and it didn't
seem like a  good idea to leave `is_muted` if sometimes the same logic was
re-implemented to combine atomic operations.

SeanTAllen reviewed (commented):
I'm a little ambivalent on removing the is muted but otherwise, this is an excellent change. I'm going to ruminate on the is muted removal.

dipinhora commented:
The main reason i removed `is_muted` is because in some cases i had to use the call to `has_sync_flag` directly to be able to combine atomic operations and it didn't seem like a good idea to leave `is_muted` if sometimes the same logic was re-implemented to combine atomic operations so i removed it for consistency (and hopefully to minimize confusion).

SeanTAllen commented:
That's an excellent reason. Can you update the first comment to include that reasoning so it makes it into the final commit message?

dipinhora commented:
updated
